.github/workflows/lint-pr.yml

```diff
@@ -0,0 +1,52 @@
+name: "Lint PR"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      # Please look up the latest version from
+      # https://github.com/amannn/action-semantic-pull-request/releases
+      - uses: amannn/action-semantic-pull-request@v3.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          # Configure which types are allowed.
+          # Default: https://github.com/commitizen/conventional-commit-types
+          # types: |
+          #   fix
+          #   feat
```

Comment:
Should we use the [default commit types](https://github.com/commitizen/conventional-commit-types/blob/master/index.json) or should we limit the commit types to the same types that we end up choosing for pull request templates (see #9341)?

Reply:
It would be nice to prefix with what the change touches. If the pr is working on staking then a prefix of `staking: <subject>` is my preferred approach. I also dont follow the need for so many templates that all say the same thing?

Reply:
> It would be nice to prefix with what the change touches.

This is the purpose of the scope. The type is what type of change is being made (i.e. feat, fix, refactor, chore, etc.). Here is an example from [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/):

```
feat(parser): add ability to parse arrays
```

> I also dont follow the need for so many templates that all say the same thing?

This could still use some fine-tuning and maybe consolidating a couple types into the same template (e.g. all the types that do not change production code). The idea with separate pull request templates is to make the checklist more relevant to each pull request (e.g. we only need a check for following the building modules documentation for feature pull requests).

Reply:
I think the default is a good starting point